src/server/game/Entities/Player/Player.cpp

```diff
@@ -14882,16 +14882,24 @@ void Player::RewardQuest(Quest const* quest, uint32 reward, Object* questGiver,
     uint32 quest_id = quest->GetQuestId();
 
     for (uint8 i = 0; i < QUEST_ITEM_OBJECTIVES_COUNT; ++i)
-        if (quest->RequiredItemId[i])
-            DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
-
+    {
+        if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(quest->RequiredItemId[i]))
+        {
+            if (quest->RequiredItemCount[i] > 0 && (itemTemplate->Bonding == BIND_QUEST_ITEM || itemTemplate->Bonding == BIND_QUEST_ITEM1))
```

Comment:
In this case its the same DestroyItemCount, what you are changing is the value passed for unequip_check (it has a default value false) - and after looking inside DestroyItemCount I can say this change is wrong